test-result-summary-client/src/TabularView/TabularView.jsx

```diff
@@ -0,0 +1,703 @@
+        + '&baselineJdkDate=' + this.state.baselineJdkDate + '&baselineJvmType=' + this.state.baselineJvmType + '&baselineJdkVersion=' + this.state.baselineJdkVersion
+        + '&baselineSdkResource=' + this.state.baselineSdkResource;
+
+        window.history.replaceState(null, '', newPath);
+    }
+    // Helper function to get value from benchmark entry
+    handleProp (val, field) {
+        if (val == null) {
+            return "N/A";
+        }
+        else if (field == 'buildUrl') {
+            let urls = {};
+            if ((val['testBuildUrl']) && (val['baselineBuildUrl'])) {
+                urls.testBuildUrl = val['testBuildUrl'];
+                urls.baselineBuildUrl = val['baselineBuildUrl'];
+            }
+            else if (val['testBuildUrl']) {
+                urls.testBuildUrl = val['testBuildUrl'];
+            } else if (val['baselineBuildUrl']) {
+                urls.baselineBuildUrl = val['baselineBuildUrl'];
+            }
+            return urls;	
+        }
+        else {return val[field];}
+    }
+
+    handleLink (urls) {
+    // Calling PerfCompare with Links
+    let url = "perfCompare?";
+    if(urls.hasOwnProperty('testBuildUrl') && urls.hasOwnProperty('baselineBuildUrl')) {
+        url+= 'testID=' + urls.testBuildUrl + '&' + 'baselineID=' + urls.baselineBuildUrl;
+    } else if (urls.hasOwnProperty('testBuildUrl')) {
+        url+= 'testID=' + urls.testBuildUrl;
+    } else if (urls.hasOwnProperty('baselineBuildUrl')) {
+        url+= 'baselineID=' + urls.baselineBuildUrl;
+    } else {return;}
+
+    const win = window.open(url, '_blank');
+    win.focus();
+    }
+    // Update table columns when platforms are ticked or unticked
+    onPlatformChange (checkedValues, event) {
+        let newArray = this.state.originalColumns;
+
+        newArray = newArray.filter(column => column.Header === "Benchmark Name" || checkedValues.includes(column.id));
+        this.setState({columns: newArray});
+    }
+    // Update table contents based on which benchmarks are ticked in the benchmark filter
+    onBenchmarkChange = value => {
+        this.setState({benchmarkFilter: value});
+        this.benchmarkFilter(value);
+    }
+    // Date picker format needs to be converted to match the one available in the database entry
+    dateTransform(date, type) {
+        const dateSplit = date.split('/');
+        const jdkDate = dateSplit[2] + ((dateSplit[0].length < 2) ? "0" + dateSplit[0] : dateSplit[0]) + ((dateSplit[1].length < 2) ? "0" + dateSplit[1] : dateSplit[1]);
+
+        if (type === 'test') {
+            this.setState({testJdkDate: jdkDate});
+        } else {
+            this.setState({baselineJdkDate: jdkDate});
+        }
+    }
+    // Based on data returned from database, create relevant columns for platforms
+    generateColumns(platforms) {
+        const newArray = [];
+        let column = {
+        Header: 'Benchmark Name',
+        accessor: 'benchmarkName',
```

Comment:
We need to give it a better name. Since `benchmarkName` just stores the benchmark name in the database, we should replace all instance of this var with something like `benchmarkNVM` (i.e. benchmark name + variant + metric).